File: Week 6/Readability/readability.py
# Import library
from cs50 import get_string

# Prompt user for input
input_text = get_string("Text: ")

# Count letters
letter_counter = 0
for letter in input_text:
    # Update counter if find alphabet
    if letter.lower() >= "a" and letter.lower() <= "z":
        letter_counter += 1

# Count words
word_counter = 0
strlen = len(input_text)
i = 0
while (i < strlen):
    # Update counter if find space(ignore multiple spaces)
    if input_text[i] == " " and input_text[i + 1] != " ":
        word_counter += 1
    i += 1
# Check containing at least one word
if strlen > 0:
    word_counter += 1
    # Ignore start with space
    if input_text[0] == " ":
        word_counter -= 1
    # Ignore end with a space
    if input_text[strlen - 1] == " ":
        word_counter -= 1

# Count sentences
sentence_counter = 0
for sentence in input_text:
    if sentence == "." or sentence == "!" or sentence == "?":
        sentence_counter += 1

# Calculate Coleman-Liau Index
L = 100 * letter_counter / word_counter
S = 100 * sentence_counter / word_counter
index = round(0.0588 * L - 0.296 * S - 15.8)

# Print result
if index >= 16:
    print("Grade 16+")
elif index < 1:
    print("Before Grade 1")
else:
    print(f"Grade {index}")

File: Week 2/ Readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <ctype.h>
#include <string.h>
#include <math.h>

// Function prototype
int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)
{
    // Getting user input
    string input_text = get_string("Text: ");

    // Calculate Coleman-Liau Index
    double L = 100.00 * count_letters(input_text) / count_words(input_text);
    double S = 100.00 * count_sentences(input_text) / count_words(input_text);
    int index = round(0.0588 * L - 0.296 * S - 15.8);

    // Output
    if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
}

// Count the number of letters
int count_letters(string text)
{
    // Keep track of letters
    int counter = 0;

    // Iterate each char
    for (int i = 0; i < strlen(text); i++)
    {
        // Update counter if find alphabet
        if (isalpha(text[i]))
        {
            counter++;
        }
    }

    return counter;
}

// Count the number of words
int count_words(string text)
{
    // Keep track of letters
    int counter = 0;

    // Iterate each char
    int len = strlen(text);
    for (int i = 0; i < len; i++)
    {
        // Update counter if find space(ignore multiple spaces)
        if (text[i] == ' ' && text[i + 1] != ' ')
        {
            counter++;
        }
    }

    // Check containing at least one word
    if (len > 0)
    {
        counter++;

        // Ignore start with space
        if (text[0] == ' ')
        {
            counter--;
        }

        // Ignore end with a space
        if (text[len - 1] == ' ')
        {
            counter--;
        }
    }

    return counter;
}

// Counter the number of sentences
int count_sentences(string text)
{
    // Keep track of sentences
    int counter = 0;

    // Iterate each char
    for (int i = 0; i < strlen(text); i++)
    {
        // Update counter if find '.', '!' or '?'
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            counter++;
        }
    }

    return counter;
}